src/quafel/pipelines/visualization/nodes.py

```diff
@@ -134,57 +279,17 @@ def shots_qubits_viz(evaluations_combined: Dict):
 
             q = int(q)
 
-            figures[f"{fw}_qubits_{q}"] = go.Figure(
-                [
-                    go.Heatmap(
-                        x=duration_sorted_by_depth["shots"].astype(int),
-                        y=duration_sorted_by_depth["depth"].astype(int),
-                        z=duration_mean,
-                        colorscale=design.seq_main,
-                        colorbar=dict(title=f"Time ({si_time})"),
-                    )
-                ]
-            )
-            figures[f"{fw}_qubits_{q}"].update_layout(
-                yaxis=dict(
-                    type=design.depth_tick_type,
-                    tickmode=design.depth_tick_mode,
-                    tickvals=duration_sorted_by_depth["depth"].astype(int)
-                    if design.depth_tick_mode == "array"
-                    else None,
-                    # ticktext=[
-                    #     f"2^{i}"
-                    #     for i in duration_sorted_by_shots["qubits"].astype(int)
-                    # ],
-                    tick0=design.depth_tick0,
-                    dtick=design.depth_dtick,
-                    title="Circuit Depth",
-                    showgrid=design.showgrid,
-                ),
-                xaxis=dict(
-                    type=design.shots_tick_type,
-                    tickmode="array",
-                    tickvals=duration_sorted_by_depth["shots"].astype(int),
-                    # ticktext=[
-                    #     f"2^{i}"
-                    #     for i in duration_sorted_by_shots["qubits"].astype(int)
-                    # ],
-                    tickangle=design.shots_tickangle,
-                    title="Num. of Shots",
-                    showgrid=design.showgrid,
-                ),
-                title=dict(
-                    text=f"{framework_name} simulation duration: Circuit Depth and Num. of Shots"
-                    if design.print_figure_title
-                    else "",
-                    font=dict(
-                        size=design.title_font_size,
-                    ),
-                ),
-                hovermode="x",
-                font=dict(
-                    size=design.legend_font_size,
-                ),
+            figures[f"{fw}_qubits_{q}"] = heatmap_viz(
+                x=duration_sorted_by_depth["shots"].astype(int),
+                y=duration_sorted_by_depth["depth"].astype(int),
+                z=duration_mean,
+                z_title=f"Time ({si_time})",
+                x_title="# of Shots",
+                log_x=True,
+                y_title="Circuit Depth",
+                log_y=True,
+                plot_title=f"{framework_name} @ {q} Qubits: "
+                f"Circuit Depth and # of Shots",
```

Comment:
Usually one needs plots without a title (e.g. for publications). So you might consider naming the plots accordingly, so that one is free to add a title with a preferred tool or to also have a flag to decide on how the title should be handled.

Reply:
agree, propose to resolve that in a separate PR -> created issue #84